app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
```

Comment:
`self.email = email.downcase` thay bằng code `email.downcase!` dc ko em

và phân biệt downcase và downcase! khác nhau thế nào nhé

Reply:
không thể thay bằng email.downcase! được vì khi người dùng nhập vào email toàn ký tự thường sẽ làm cho email trả về kết quả nil
downcase : chuyển chuỗi về ký tự thường nhưng biến ban đầu không thay đổi kết quả
downcase! : chuyển chuỗi về ký tự thường và thay đổi kết quả của biến, khi chuyển một chuỗi toàn ký tự thường sẽ trả về nil

Reply:
tồ lô, sửa nhé
thay nguyên dòng code `self.email = email.downcase` thành `email.downcase!` nhé